From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct perf_event_context
Patch-mainline: never, kabi
References: kabi

In 3.12.68, commit 5e08a111b0a076648039fb2a08d6e101a6af9388 (perf:
Tighten (and fix) the grouping condition), upstream commit
c3c87e770458aa004bd7ed3f29945ff436fd6511
removed enum perf_event_context_type and its presence in struct
perf_event_context. This changed almost everything in the kernel,
given perf_event_context is embedded in task_struct.

So re-declare the enum and add a dummy member back to struct
perf_event_context to make the kABI checker happy.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/perf_event.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/linux/perf_event.h
+++ b/include/linux/perf_event.h
@@ -453,6 +453,11 @@ struct perf_event {
 #endif /* CONFIG_PERF_EVENTS */
 };
 
+enum perf_event_context_type {
+	task_context,
+	cpu_context,
+};
+
 /**
  * struct perf_event_context - event context structure
  *
@@ -460,6 +465,7 @@ struct perf_event {
  */
 struct perf_event_context {
 	struct pmu			*pmu;
+	enum perf_event_context_type	type;
 	/*
 	 * Protect the states of the events in the list,
 	 * nr_active, and the list:
